src/SonarScanner.MSBuild.Common/OperatingSystemProvider.cs

```diff
@@ -71,4 +73,18 @@ private static PlatformOS CurrentOperatingSystem()
 #else
         File.Exists("/System/Library/CoreServices/SystemVersion.plist");
 #endif
+
+    // Not testable due to file system access. Manual testing was done by running the scanner on Alpine.
+    [ExcludeFromCodeCoverage]
+    private static bool IsAlpine() =>
+        IsAlpineRelease("/etc/os-release")
+        || IsAlpineRelease("/usr/lib/os-release");
+
+    // See: https://www.freedesktop.org/software/systemd/man/latest/os-release.html
+    // Examples: "ID=alpine", "ID=fedora", "ID=debian".
+    // Not testable due to file system access. Manual testing was done by running the scanner on Alpine.
+    [ExcludeFromCodeCoverage]
+    private static bool IsAlpineRelease(string releaseInfoFilePath) =>
+        File.Exists(releaseInfoFilePath)
+        && File.ReadAllText(releaseInfoFilePath).Contains("ID=alpine");
```

Comment:
We do have `IFileWrapper` to mock `File.Exists` and `ReadAllText`. I used it in #1981 already, so there might be merge conflicts arising from its use here (in classes like the ArgumentProcessor). With that wrapper, we can test this with the help of NSubstitute.

Reply:
I will rebase on that PR to avoid merge conflicts.

Reply:
I had to rework a bit how the IOperatingSystemProvider was implemented but I think the new approach will be more flexible and will help when implementing https://github.com/SonarSource/sonar-scanner-msbuild/issues/1988